src/network/networkd-network.h

```diff
@@ -325,6 +325,7 @@ struct Network {
         int ipv6_dad_transmits;
         uint8_t ipv6_hop_limit;
         int proxy_arp;
+        int proxy_arp_pvlan;
```

Comment:
so it his option actually really orthogonal to regular proxy arp?

Reply:
It subsumes the proxy_arp one, if you enable proxy_arp_pvlan, you don't need regular proxy ARP.

Reply:
So if that's really the case then it probably should not be exposed like that, but instead the existing option should just become an enum that besides yes/no also accepts "pvlan" or so as value to enable this mode.

So looking at the implementation, the feature might actually be orthogonal after all though:

https://patchwork.ozlabs.org/project/netdev/patch/20100105155047.13309.79610.stgit@firesoul.comx.local/

i.e. if you look at that, then classic proxy arp only forwards between *different* network devices, while this new thing only forwards from an interface onto itself. Thus you can enable one or the other, or both, and each time behevaiour changes.

but maybe i am misreading the code. 

but if i am right, then two distinct options like your current patch probably do make some sense.

Reply:
Hm, you are absolutely right, rereading this.

Reply:
Let me know if I should do something here, resolving in the meantime.